File: 2025/day01.py
from pathlib import Path

moves = [
    (line[0], int(line[1:]))
    for line in Path("./day01.txt").read_text().split("\n")
    if line
]

result = 50
zeros = 0

for direction, count in moves:
    if direction == "L":
        result -= count
    else:
        result += count

    if result % 100 == 0:
        zeros += 1

print(zeros)

result = 50
zeros = 0
for direction, count in moves:
    if direction == "L":
        new_result = result - count
        if new_result % 100 == 0 or (
            result % 100 > 0 and new_result % 100 > result % 100
        ):
            zeros += 1
    else:
        new_result = result + count
        if new_result % 100 < result % 100:
            zeros += 1

    zeros += count // 100
    # print(f"{result} + {direction}{count} -> {new_result}, {zeros}")
    result = new_result

print(zeros)

File: 2025/day01.go
package main

import "github.com/erdnaxeli/adventofcode/aoc"

func (s solver) Day1p1(input aoc.Input) string {
	// Direct copy of the python implementation.
	zeros := 0
	position := 50

	for _, line := range input.ToStringSlice() {
		direction := line.At(0)
		count := line.From(1).Atoi()

		if direction == 'L' {
			position -= count
		} else {
			position += count
		}

		if position%100 == 0 {
			zeros++
		}
	}

	return aoc.ResultI(zeros)
}

func (s solver) Day1p2(input aoc.Input) string {
	return ""
}
